### PR TITLE
Remove unused code

### PR DESCRIPTION
**Pull request by enricosada**
simplify codebase, remove code inside unused directives

each commit is one directive, so easier to rollback or discard

there are lot of directives (160+):

### old stuff
obsolete -> remove?

- OLD_CACHE
- WORKAROUND_FSHARP_2_0_BUG

### features wip/disabled/incomplete
remove or move to branch/pr?

- DESIGNER
- TurnWordWrapOnByDefault
- COLORABLE_ITEM
- COLORIZE_TYPES

### not used?
not only the silvelight stuff

- FEATURE_PAL
- !PLATFORM_UNIX
- `__APPLE__`

### debugger helpers
use a naming convention?

- DEBUGGERVISUALIZER
- __DEBUG
- EXTRA_DEBUG

and others..

### CODE REVIEW

File: src/fsharp/lib.fs [hunk 1]
```diff
@@ -383,53 +383,6 @@ type Graph<'Data, 'Id when 'Id : comparison and 'Id : equality>
             else List.iter (trace (node.nodeData::path)) node.nodeNeighbours
         List.iter (fun node -> trace [] node) nodes 
 
-#if OLDCODE
```
Comment (forki) [inline src/fsharp/lib.fs]: ?

Comment (enricosada) [inline src/fsharp/lib.fs]: exactly

File: vsintegration/src/vs/FsPkgs/FSharp.Project/Common.Source.CSharp/Project/FileNode.cs [hunk 2]
```diff
@@ -1122,47 +1067,7 @@ private void RenameCaseOnlyChange(string newFileName)
 
         #endregion
 
-#region SingleFileGenerator Support methods
```
Comment (dsyme) [inline vsintegration/src/vs/FsPkgs/FSharp.Project/Common.Source.CSharp/Project/FileNode.cs]: I'm not sure about deleting the single-file-generator stuff, someone may want to take a stab at enabling this feature in the project system.  I'll let @latkin and @KevinRansom decide

**Comment by dsyme** [comment]
Great work

**Comment by dsyme** [comment]
Thank you for doing this - this kind of spring cleaning is so important to maintaining the viability of the codebase going forward, I really appreciate you doing this.

File: vsintegration/src/vs/FsPkgs/FSharp.Project/Common.Source.CSharp/Project/FileNode.cs (hunk 2 above)
Comment (enricosada) [inline vsintegration/src/vs/FsPkgs/FSharp.Project/Common.Source.CSharp/Project/FileNode.cs]: wip features inside of directives can be moved to a feature branch.

I can:

- create a branch singleFileGenerator
- revert this commit in stage area
- remove the `#if` `#endif`
- commit 

now the singleFileGenerator branch contains all the code **active**, so easy diff (all code is a single commi) and rebase

If the codebase change, i dont think there is a build matrix with all the directives, maybe some are not aligned or do not compile anymore (this is the problem with conditional directives). 
If someone want to continue the work, can easly use this new branch

**Comment by KevinRansom** [comment]
Thank you for taking the time to take care of this work, we will take care of this at the start of the next cycle.

Moving this to V.Next per: #371

**Comment by latkin** [comment]
Applied to the `OOB` branch.

**Pull request closed without merging**